### PR TITLE
Network tests

### LINKED ISSUE #25: Add network.spec.js tests
Based on this tag https://github.com/microsoft/playwright/tree/v0.10.0. We don't need to get the tests green. We only need to incorporate the code. Let's also check if we can get code from Puppeteer

### PR DESCRIPTION
**Pull request by kblok**
closes #25

### CODE REVIEW

File: src/PlaywrightSharp.Abstractions/IResponse.cs [hunk 1]
```diff
@@ -49,5 +50,17 @@ public interface IResponse
         /// </summary>
         /// <returns>A <see cref="Task"/> that completes when the text was processed, yielding to a text representation of response body.</returns>
         Task<string> GetTextAsync();
+
+        /// <summary>
+        /// Returns a <see cref="Task"/> which resolves to a <see cref="JsonDocument"/> representation of response body.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> that completes when the json body is parsed, yielding a <see cref="JsonDocument"/> representation of response body.</returns>
+        Task<JsonDocument> GetJsonAsync();
```
Comment (Meir017) [inline src/PlaywrightSharp.Abstractions/IResponse.cs]: don't we want this to return `T`?

File: src/PlaywrightSharp.Abstractions/IResponse.cs [hunk 2]
```diff
@@ -49,5 +50,17 @@ public interface IResponse
...
+        /// Returns a <see cref="Task"/> which resolves to a <see cref="JsonDocument"/> representation of response body.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> that completes when the json body is parsed, yielding a <see cref="JsonDocument"/> representation of response body.</returns>
+        Task<JsonDocument> GetJsonAsync();
+
+        /// <summary>
+        /// Returns a <see cref="Task"/> which resolves to a buffer with response body.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> that completes when the response is returned by the server, yielding a <see cref="byte"/> array.</returns>
+        Task<byte[]> GetBufferAsync();
```
Comment (Meir017) [inline src/PlaywrightSharp.Abstractions/IResponse.cs]: can this be `Span<byte>`?

File: src/PlaywrightSharp.Tests/PlaywrightSharp.Tests.csproj [hunk 3]
```diff
@@ -5,6 +5,7 @@
         <IsPackable>false</IsPackable>
         <IsTestProject>true</IsTestProject>
         <LangVersion>8.0</LangVersion>
+        <ReleaseVersion>0.0.0</ReleaseVersion>
```
Comment (Meir017) [inline src/PlaywrightSharp.Tests/PlaywrightSharp.Tests.csproj]: is this needed?

File: src/PlaywrightSharp.Abstractions/IResponse.cs (hunk 1 above)
Comment (kblok) [reply src/PlaywrightSharp.Abstractions/IResponse.cs]: we could do both :)

File: src/PlaywrightSharp.Abstractions/IResponse.cs (hunk 2 above)
Comment (kblok) [reply src/PlaywrightSharp.Abstractions/IResponse.cs]: What would be the benefit for the consumer?

File: src/PlaywrightSharp.Tests/PlaywrightSharp.Tests.csproj (hunk 3 above)
Comment (kblok) [reply src/PlaywrightSharp.Tests/PlaywrightSharp.Tests.csproj]: None of the changes here

File: src/PlaywrightSharp.Abstractions/IResponse.cs (hunk 2 above)
Comment (kblok) [reply src/PlaywrightSharp.Abstractions/IResponse.cs]: I just checked that you can implicitly cast T[] to Span<T> ([source](https://docs.microsoft.com/en-us/archive/msdn-magazine/2018/january/csharp-all-about-span-exploring-a-new-net-mainstay).

I think a byte[] as a more pure and generic type.

**Pull request merged**